src/ngx_http_lua_util.h

```diff
@@ -379,7 +379,8 @@ ngx_http_lua_set_content_type(ngx_http_request_t *r)
     ngx_http_lua_loc_conf_t     *llcf;
 
     llcf = ngx_http_get_module_loc_conf(r, ngx_http_lua_module);
-    if (llcf->use_default_type) {
+    if (llcf->use_default_type
+            && r->headers_out.status != NGX_HTTP_NOT_MODIFIED) {
```

Comment:
Style: nginx usually aligns `&&` with the first character after `(` in the previous line. Also, for splitted `if` statements, nginx usually puts the `{` on a separate line (and aligns it up with `if`'s beginning column).